sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/Offset.scala

```diff
@@ -33,5 +35,17 @@ trait Offset extends Serializable {
   def <(other: Offset): Boolean = compareTo(other) < 0
   def <=(other: Offset): Boolean = compareTo(other) <= 0
   def >=(other: Offset): Boolean = compareTo(other) >= 0
-  def ==(other: Offset): Boolean = compareTo(other) == 0
+
+  override def equals(other: Any): Boolean = {
```

Comment:
I'm not sure that its valid to override `equals` without also overriding `hashCode`